### DISCUSSION TITLE
Duda sobre estructuras

### DISCUSSION THREAD

**Discussion by LautaroUlloa** [INGCOM-UNRN-P1/cursada-2023#197] [Preguntas y respuestas] [answered]
Estaba experimentando con estructuras y me saltó un error que no estoy entendiendo:

Primero que nada defino la estructura:
```C
typedef struct cuenta
{
    char nombre[30];
    char apellido[40];
    int saldo;
} cuenta_t;
```
El problema es cuando quiero darle valores a las variables de la estructura en el main. Cuando hago algo como esto
```C
int main()
{
    cuenta_t c1;
    c1.nombre = "Pepito";
    ...
    return 0;
}
```
Me salta un error que dice "error: assignment to expression with array type".

Pero si lo hago de cualquiera de las siguientes maneras, me anda bien:

```scanf("%s", &c1.nombre);```
```cuenta_t c1 = {"Pepito"...};```
```cuenta_t c1 = {.nombre = "Pepito"...};```

No entiendo el error.

**Comment by TeoInMaghine** [comment, marked as answer]
No tiene que ver con que uses un struct, sino porque solo se puede asignar un string (arreglo de caracteres) de esa forma cuando lo declarás. Si hacés `nombre = "Pepito";` se queja el compilador porque espera que le asignes una dirección de memoria/puntero. Si querés asignarlo con un valor después de haberlo declarado podés asignar por índice (`nombre[0] = 'P';`) carácter a carácter, o usar strcpy:
```C
#include <stdio.h>
#include <string.h>

int main()
{
    char nombre[30];    
    strcpy(nombre, "Pepito");
    printf("%s\n", nombre);
}
```

**Reply by LautaroUlloa** [reply]
Ah, buenísimo. Muchas gracias

**Comment by martinvilu** [comment]
En la clase de hoy (11/10) tratamos tambien este tema para mas detalles.
